### PR TITLE
Add cookies injection

### PR DESCRIPTION
**Pull request by ShaharHD**
Add the ability to inject cookies (useful for automatic logins with tokens)

### CODE REVIEW

**Comment by ShaharHD** [comment]
hello? :)

File: README.md [hunk 1]
```diff
@@ -18,18 +18,20 @@ Watch me create a Slack clone from scratch in 30 seconds (<a href="https://kofi.
...
+- [Using the CLI: `create-mac-app`](#using-the-cli-create-mac-app)
+- [Custom JS/CSS](#custom-jscss)
+    - [Fix links in GMail and Google Calendar](#fix-links-in-gmail-and-google-calendar)
+    - [Reload Slack when it disconnects](#reload-slack-when-it-disconnects)
+    - [Find in page](#find-in-page)
+    - [Drag-and-drop to open URLs](#drag-and-drop-to-open-urls)
+    - [Preview link targets](#preview-link-targets)
+- [Custom Cookies](#custom-cookies)
+- [Keyboard shortcuts](#keyboard-shortcuts)
+- [Donating](#donating)
```
Comment (kofigumbs) [inline README.md:34]: Would you mind reverting the unnecessary changes here:

- Leave indentation as-is
- Remove the "Table of contents" entry

File: Sources/Runtime/Config.swift [hunk 2]
```diff
@@ -6,16 +6,33 @@ struct Config {
         let alwaysNotify: Bool?
         let openNewWindowsWith: String?
         let openNewWindowsInBackground: Bool?
+        let customCookie: [Config.Schema.Cookie]?
```
Comment (kofigumbs) [inline Sources/Runtime/Config.swift:9]: I'd like to keep this config exclusively at the tab level.

File: README.md [hunk 3]
```diff
@@ -72,6 +74,7 @@ The `tabs` field is an array of objects with the following fields:
 | `url`               | String (Required)           | Starting page for this tab                                             |
 | `customJs`          | Array of Strings (Optional) | Custom JS URLs (see [Custom JS/CSS](#custom-jscss))                    |
 | `customCss`         | Array of Strings (Optional) | Custom CSS URLs (see [Custom JS/CSS](#custom-jscss))                   |
+| `customCookie`      | Array of Cookies (Optional) | Custom Cookies injection (see [Custom Cookies](#custom-cookies))       |
```
Comment (kofigumbs) [inline README.md:77]: Can we call the field `customCookies` instead (plural)?

Also, I think it needs its own documentation table, like we have for `tabs`. I'd prefer if users didn't need to read WebKit docs to understand Multi settings.

**Review by kofigumbs — COMMENTED** [review]
This is rad, thanks! I left a few comments, for whenever you get some time.

File: README.md (hunk 1 above)
Comment (ShaharHD) [reply README.md]: Done

File: Sources/Runtime/Config.swift (hunk 2 above)
Comment (ShaharHD) [reply Sources/Runtime/Config.swift:9]: There's a reason I've added this on the global level as well.
The cookie injection is used in many instances where auto login is problematic (like Grafana) - and the logic token keep expiring.
If it has to be on the tab level, this means every time it expires - you'll have to manully update it for each tab and not once on the top level.

File: README.md (hunk 3 above)
Comment (ShaharHD) [reply README.md]: Done

Comment (ShaharHD) [reply README.md]: The documention is more complex than we can simply hold in a table as there some dependency between some of the parameters.
One example is `version` and `discard`

If `version` is `0` (the default) then the default for `discard` is `FALSE` and otherwise it's `TRUE`

how do you want to approach this?

File: Sources/Runtime/Config.swift (hunk 2 above)
Comment (kofigumbs) [reply Sources/Runtime/Config.swift]: In that scenario, I'd recommend using a separate tool (like a Bash script) to template/generate your JSON config. That's one of the advantages of having the config file live as a plain JSON file on disk.

File: README.md (hunk 3 above)
Comment (kofigumbs) [reply README.md]: Good point, I left the link. On further reflection, I think this feature will be rarely used, so the entire table seemed like overkill.

**Comment by kofigumbs** [comment]
I merged a version of this in 5219804. Thanks again.

File: Sources/Runtime/Config.swift (hunk 2 above)
Comment (ShaharHD) [reply Sources/Runtime/Config.swift]: most of the users I've been installing the took on for the graphana. ... are not that advanced...
they just know to replace the token key in one location (they get it in the email)

I'll just keep my version for now - as we disagree on this when I prefer to have the ability to set global cookies (and allow to override them on a tab basis if needed)
